src/doc/unstable-book/src/language-features/register-tool.md

```diff
@@ -0,0 +1,48 @@
+# `register_tool`
+
+The tracking issue for this feature is: [#66079]
+
+[#66079]: https://github.com/rust-lang/rust/issues/66079
+
+------------------------
+
+The `register_tool` language feature informs the compiler that attributes in your code are meant to be used with tools other than the compiler itself. This can be useful if your code has semantic meaning without the external tool, but enables additional features when the tool is present.
+
+`register_tool` also allows configuring lint levels for external tools.
+
+Tool attributes are only meant for ignorable attributes. If your code *changes* meaning when the attribute is present, it should not use a tool attribute (because it cannot be compiled with anything other than the external tool, and in a sense is a fork of the language).
+
+------------------------
+
+`#![register_tool(tool)]` is an attribute, and is only valid at the crate root.
+Attributes using the registered tool are checked for valid syntax, and lint attributes are checked to be in a valid format. However, the compiler cannot validate the semantics of the attribute, nor can it tell whether the configured lint is present in the external tool.
+
+Semantically, `clippy::*`, `rustdoc::*`, and `rustfmt::*` lints and attributes all behave as if `#![register_tool(clippy, rustdoc, rustfmt)]` were injected into the crate root, except that the `rustdoc` namespace can only be used for lints, not for attributes.
+When compiling with `-Z unstable-features`, `rustc::*` lints can also be used. Like `rustdoc`, the `rustc` namespace can only be used with lints, not attributes.
```

Comment:
Is there a tracking issue for this stabilization? That sort of backlink is really helpful for catching when unstable features are eventually stabilized and updating the docs.

Reply:
The original tracking issue is https://github.com/rust-lang/rust/issues/66079, but it's probably best if a separate tracking issue is opened for the stabilization because that original issue is really hard to follow.

Reply:
register_tool is tracked in https://github.com/rust-lang/rust/issues/66079. `rustc::` lints are mega unstable and in general a mess; they're really only used by bootstrap.

Reply:
Yeah, I was asking about the rustc:: lints :) Annoying that those are perma-unstable, but if that's the case there's really not any point linking something.

Reply:
I don't think `rustc` lints will ever be stabilized, AFAIK they are *really* internal lints for compiler codebase itself.